### PR TITLE
several bugfixes and general improvements, see commits below.

### PR DESCRIPTION
**Pull request by R41z**

### CODE REVIEW

**Comment by drkhsh** [comment]
Thank you for your great changes! Could you squash some commits to clean up a bit? Like the "whoops" commits etc 😄

**Comment by R41z** [comment]
sorry, I don't know how to squash these :/

**Comment by drkhsh** [comment]
@R41z nevermind, not that important 😄

File: slstatus.c [hunk 1]
```diff
@@ -759,7 +759,7 @@ int
 main(int argc, char *argv[])
 {
 	unsigned short int i;
-	char status_string[4096];
+	char status_string[2048];
```
Comment (drkhsh) [inline slstatus.c]: Why 2048 here? What if users (like me) want to use longer status bars? 😄 I don't think that 4096 is too much.

File: slstatus.c [hunk 2]
```diff
@@ -811,7 +811,8 @@ main(int argc, char *argv[])
 				element = smprintf(UNKNOWN_STR);
 				warnx("Failed to format output");
 			}
-			strlcat(status_string, element, sizeof(status_string));
+			if (strlcat(status_string, element, sizeof(status_string)) >= sizeof(status_string))
+				warnx("Output too long");
```
Comment (drkhsh) [inline slstatus.c]: My coding style is to use braces also for single-line statements. Could you please add these?

File: TODO.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+[ ] fix build on OpenBSD
+[ ] fix build on NetBSD
+[ ] fix build on FreeBSD
+[ ] test wifi functions after the last few commits
```
Comment (drkhsh) [inline TODO.md]: It does **not** work 😭

Comment (drkhsh) [inline TODO.md]: oops, it works 😸

**Comment by drkhsh** [comment]
Fuck this shit. The changes above ok? :smile:

**Comment by R41z** [comment]
I thought you follow the suckless style :)

File: slstatus.c (hunk 1 above)
Comment (R41z) [inline slstatus.c]: not really, a status bar on a 1920x1080 monitor would not even need that 2048, see: http://hexeract.info/pub/len.png

**Pull request merged**